api/objects.proto

```diff
@@ -122,6 +122,12 @@ message Service {
 	// UpdateStatus contains the status of an update, if one is in
 	// progress.
 	UpdateStatus update_status = 5;
+
+	// services, as well as all service-level resources, can only be deleted
+	// after all the service's containers have properly shut down
+	// when a user requests a deletion, we just flip this flag
+	// the deallocator will take it from there
+	bool pending_delete = 7;
```

Comment:
nit: comments in protobufs should be in the form like `// PendingDelete indicates...`, which causes the resulting generated code to have godoc style comments.